src/sdsc/spellcheck.py

```diff
@@ -0,0 +1,232 @@
+            if not word:
+                continue
+            # filter for numbers, punctuation, URLs, file names
+            if SPELLFILTER.match(word):
+                continue
+
+            correct = spelldict.check(word)
+
+            if not correct:
+                # FIXME: can we order suggestions, e.g. for "ipc" (which is in
+                # our dictionary as "IPC"), we only get irrelevant lower-case
+                # suggestions and "IPC" removed later when we limit the choice
+                # to 5 suggestions.
+                suggestions = spelldict.suggest(word)
+
+                line = linenumber(context)
+                contenthighlighted = highlightREMOVEME(xmlescape(contentpretty), currenttokeninparagraph, currenttokeninparagraph)
+                messages.append(spellcheckmessage(
+                    suggestions, word, line,
+                    contenthighlighted, contextid, basefile,
+                    messagetype))
+
+    return messages
+
+
+def spellcheckmessage(suggestions, word, line, content,
+                        contextid, basefile, messagetype):
+    """ Create a message after a spell check has found a non-dictionary word.
+
+    :param list suggestions: list of strings with suggestions
+    :param str word: spelling, as actually used
+    :param int line: line number (generally wrong at the moment)
+    :param str content: paragraph of text that "word" appears in
+    :param str contextid: value of id attribute on next node
+    :param str basefile: name of file that "word" appears in
+    :param str messagetype: print 'error', 'warning', or 'info' message?
+    """
+
+    # FIXME: shorten content string (in the right place), to get closer toward
+    # more focused results
+    message = None
+    filename = ""
+    if basefile:
+        filename = "<file>%s</file>" % str(basefile)
+
+    withinid = ""
+    if contextid:
+        withinid = "<withinid>%s</withinid>" % str(contextid)
+
+    message = etree.XML("""<result type="%s">
+            <location>%s%s<line>%s</line></location>
+        </result>""" % (messagetype, filename, withinid, str(line)))
+
+    message.append(etree.XML("""<message>Do not use
+        <quote>%s</quote>:
+        <quote>%s</quote></message>""" % (word, content)))
+
+    if suggestions:
+        # Sometimes enchant will give eight or ten suggestions. This clutters
+        # the view quite a bit, so show at most 5 suggestions -- still not
+        # quite sure if this is a good idea, though: Sometimes relevant
+        # suggestions are removed now.
+        suggestions = suggestions[:5]
+        for suggestion in suggestions:
+            message.append(etree.XML("""<suggestion>Correct to
+                <quote>%s</quote>.</suggestion>""" % suggestion))
+
+    return message
+
+#### REMOVE EVERYTHING BELOW WHEN I KNOW HOW TO PROPERLY IMPORT FROM MAIN!
+
+def highlightREMOVEME(tokens, highlightstart, highlightend):
```

Comment:
Very creative naming of this function. 😉 I don't think this is PEP8 conform. Maybe use another name which is conforms more with PEP8.

Reply:
as the comment two lines above it says: this is copypasta from __init__ which i need to get rid of before merging anyway.